packages/next/src/build/index.ts

```diff
@@ -3705,6 +3712,18 @@ export default async function build(
 
       await shutdownPromise
     })
+  } catch (e) {
```

Comment:
We might need to fix some usage of `process.exit(1)` above to catch more cases but we can investigate that separate